src/LLVM_Output.cpp

```diff
@@ -28,7 +28,7 @@ std::unique_ptr<llvm::raw_fd_ostream> make_raw_fd_ostream(const std::string &fil
 
 
 #if LLVM_VERSION < 37
-void emit_file_legacy(llvm::Module &module, const std::string &filename, llvm::TargetMachine::CodeGenFileType file_type) {
+void emit_file_legacy(llvm::Module &module, llvm::raw_fd_ostream& out, llvm::TargetMachine::CodeGenFileType file_type) {
```

Comment:
I think most of these should take an llvm::raw_ostream, not raw_fd_ostream, so they can be used with in-memory buffers or other raw_ostream objects.